src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,85 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
+/**
+ * If the class is annotated with the @Component annotation, then we can create an implementation
+ * If not, throw an error
+ * Check if an instance has been created before creating a new instance
+ */
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
+    private Map<Class<?>, Class<?>> interfaceImplementations;
```

Comment:
This one should be hardcoded 
![image](https://github.com/mate-academy/jv-dependency-injection/assets/69054906/9ca09526-049f-49ec-915b-072c2d65db8b)